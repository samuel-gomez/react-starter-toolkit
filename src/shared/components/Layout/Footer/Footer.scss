.af-footer {
  &--fullscreen {
    .container {
      max-width: 100%;
    }
  }
}
