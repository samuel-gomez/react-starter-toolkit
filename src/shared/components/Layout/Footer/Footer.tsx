import Image from 'next/image';
import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import packageJson from '../../../../../package.json';
import './Footer.scss';

type TFooterAxa = {
  version?: string;
  icon?: string;
  copyright?: string;
  year?: number;
  fullScreen?: boolean;
};

const FooterCustom = ({ className, icon, copyright }: { className: string; icon: string; copyright: string }) => (
  <footer className={className}>
    <div className="container-fluid container">
      <a className="af-logo" href="https://www.axa.fr/" title="Site Axa" target="blank">
        <Image className="af-logo__brand" alt="Logo Axa" src={icon} width={22} height={22} />
      </a>
      <div className="af-footer-content">{copyright}</div>
    </div>
  </footer>
);

const FooterAxa = ({
  fullScreen,
  year = new Date().getFullYear(),
  version = packageJson.version,
  icon = logo.src,
  copyright = `© ${year} AXA Tous droits réservés - v${version}`,
}: TFooterAxa) => <FooterCustom className={`af-footer${fullScreen ? ' af-footer--fullscreen' : ''}`} icon={icon} copyright={copyright} />;

export default FooterAxa;
