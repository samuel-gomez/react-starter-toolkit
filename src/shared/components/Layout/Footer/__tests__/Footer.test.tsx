import { render } from '@testing-library/react';
import Footer from '../Footer';

it('Render <Footer/>', () => {
  const { asFragment } = render(<Footer />);
  expect(asFragment()).toMatchSnapshot();
});

it('Render <Footer/> fullscreen', () => {
  const { asFragment } = render(<Footer fullScreen />);
  expect(asFragment()).toMatchSnapshot();
});
