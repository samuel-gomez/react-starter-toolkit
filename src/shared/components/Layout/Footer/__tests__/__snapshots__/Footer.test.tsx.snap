// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Footer/> 1`] = `
<DocumentFragment>
  <footer
    class="af-footer"
  >
    <div
      class="container-fluid container"
    >
      <a
        class="af-logo"
        href="https://www.axa.fr/"
        target="blank"
        title="Site Axa"
      >
        <img
          alt="Logo Axa"
          class="af-logo__brand"
          data-nimg="1"
          decoding="async"
          height="22"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="22"
        />
      </a>
      <div
        class="af-footer-content"
      >
        © 2023 AXA Tous droits réservés - v1.0.0
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Render <Footer/> fullscreen 1`] = `
<DocumentFragment>
  <footer
    class="af-footer af-footer--fullscreen"
  >
    <div
      class="container-fluid container"
    >
      <a
        class="af-logo"
        href="https://www.axa.fr/"
        target="blank"
        title="Site Axa"
      >
        <img
          alt="Logo Axa"
          class="af-logo__brand"
          data-nimg="1"
          decoding="async"
          height="22"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="22"
        />
      </a>
      <div
        class="af-footer-content"
      >
        © 2023 AXA Tous droits réservés - v1.0.0
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
